src/api/fuel.js

```diff
@@ -314,44 +313,47 @@ async function confirmWithUser(user/*: User*/, fees/*: string | null*/) {
         // this are the normal texts for random wallet.
         const cancel/*: string | boolean*/ = GreymassFuelService.globals.$t('api.reject');
         const ok = GreymassFuelService.globals.$t('api.confirm');
-        let message = GreymassFuelService.globals.$t('api.greymass_fuel_message');
 
         if (typeof fees === 'string') {
-            message = GreymassFuelService.globals.$t('api.greymass_fuel_message_fees', { fees });
+            // Only if there's some feed to charge the user wi show the confirmation Dialog
+            const message = GreymassFuelService.globals.$t('api.greymass_fuel_message_fees', { fees });
+            Dialog.create({
+                title: GreymassFuelService.globals.$t('api.greymass_dialog_title'),
+                message,
+                html: true,
+                cancel,
+                ok,
+                persistent: true,
+                class: 'text-black',
+                options: {
+                    type: 'checkbox',
+                    model: mymodel,
+                    isValid: (model/*: string | string[]*/) => {
+                        GreymassFuelService.setPreferences(username, {
+                            remember: model.length === 1,
+                        });
+                        return true;
+                    },
+                    items: [
+                        {
+                            label: GreymassFuelService.globals.$t('api.remember_my_decision'),
+                            value: 'remember',
+                            color: 'primary' },
+                    ],
+                },
+            })
+            // all answers should save the preferences
+                .onOk(() => handler(true))
+                .onCancel(() => handler(false));
```

Comment:
We only present this Dialog in the case of having any extra fees that the user should approve to pay. If there are no fees, then no Dialog is shown to the user.